diff --git a/chirp/ui/mainapp.py b/chirp/ui/mainapp.py
index 079f60f..0e89d98 100644
--- a/chirp/ui/mainapp.py
+++ b/chirp/ui/mainapp.py
@@ -906,9 +906,9 @@ of file.
 
         # Do this in case the import process is going to take a while
         # to make sure we process events leading up to this
-        gtk.gdk.window_process_all_updates()
+        gtk.gdk.Window.process_all_updates()
         while gtk.events_pending():
-            gtk.main_iteration(False)
+            gtk.main_iteration_do(False)
 
         if do_import:
             eset = self.get_current_editorset()
@@ -1035,9 +1035,9 @@ of file.
 
         # Do this in case the import process is going to take a while
         # to make sure we process events leading up to this
-        gtk.gdk.window_process_all_updates()
+        gtk.gdk.Window.process_all_updates()
         while gtk.events_pending():
-            gtk.main_iteration(False)
+            gtk.main_iteration_do(False)
 
         fn = tempfile.mktemp(".csv")
         filename, headers = urllib.urlretrieve(query, fn)
@@ -1141,9 +1141,9 @@ of file.
 
         # Do this in case the import process is going to take a while
         # to make sure we process events leading up to this
-        gtk.gdk.window_process_all_updates()
+        gtk.gdk.Window.process_all_updates()
         while gtk.events_pending():
-            gtk.main_iteration(False)
+            gtk.main_iteration_do(False)
 
         fn = tempfile.mktemp(".csv")
         filename, headers = urllib.urlretrieve(query, fn)
@@ -1311,9 +1311,9 @@ of file.
 
         # Do this in case the import process is going to take a while
         # to make sure we process events leading up to this
-        gtk.gdk.window_process_all_updates()
+        gtk.gdk.Window.process_all_updates()
         while gtk.events_pending():
-            gtk.main_iteration(False)
+            gtk.main_iteration_do(False)
 
         if do_import:
             eset = self.get_current_editorset()
@@ -1374,9 +1374,9 @@ of file.
 
         # Do this in case the import process is going to take a while
         # to make sure we process events leading up to this
-        gtk.gdk.window_process_all_updates()
+        gtk.gdk.Window.process_all_updates()
         while gtk.events_pending():
-            gtk.main_iteration(False)
+            gtk.main_iteration_do(False)
 
         if do_import:
             eset = self.get_current_editorset()
