diff --git a/chirp/drivers/uv5r.py b/chirp/drivers/uv5r.py
index 18fff8b..4ce127e 100644
--- a/chirp/drivers/uv5r.py
+++ b/chirp/drivers/uv5r.py
@@ -579,7 +579,7 @@ def _do_download(radio):
             data += _read_block(radio, i, 0x40, False)
 
     if append_model:
-        data += radio.MODEL.ljust(8)
+        data += radio.MODEL.ljust(8).encode()
 
     LOG.debug("done.")
     return memmap.MemoryMap(data)
diff --git a/chirp/drivers/uvb5.py b/chirp/drivers/uvb5.py
index e4d53ba..161c970 100644
--- a/chirp/drivers/uvb5.py
+++ b/chirp/drivers/uvb5.py
@@ -207,7 +207,7 @@ def do_download(radio):
     data += ("\x00" * 16)
     firstack = None
     for i in range(0, 0x1000, 16):
-        frame = struct.pack(">cHB", "R", i, 16)
+        frame = struct.pack(">cHB", b"R", i, 16)
         radio.pipe.write(frame)
         result = radio.pipe.read(20)
         if frame[1:4] != result[1:4]:
